src/js/player.js

```diff
@@ -1528,20 +1528,19 @@ class Player extends Component {
       // if the `sourceset` `src` was an empty string
       // wait for a `loadstart` to update the cache to `currentSrc`.
       // If a sourceset happens before a `loadstart`, we reset the state
-      // as this function will be called again.
       if (!event.src) {
-        const updateCache = (e) => {
-          if (e.type !== 'sourceset') {
-            const techSrc = this.techGet('currentSrc');
-
-            this.lastSource_.tech = techSrc;
-            this.updateSourceCaches_(techSrc);
+        this.tech_.any(['sourceset', 'loadstart'], (e) => {
+          // if a sourceset happens before a `loadstart` there
+          // is nothing to do as this function will be called again.
+          if (e.type === 'sourceset') {
```

Comment:
Using `.any`, I'm not sure that this function will get called again anymore.

Reply:
I think the comment needs to be updated to say that `handleTechSourceset` will be called again and reset the cache, rather than "this function"

Reply:
Oh, I see, we called `off` unconditionally previously as well. It looked like we didn't.

Reply:
I updated the comment now.

Reply:
Thanks!